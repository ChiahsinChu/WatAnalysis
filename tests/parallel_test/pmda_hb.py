# SPDX-License-Identifier: LGPL-3.0-or-later
import time

import MDAnalysis as mda
from MDAnalysis import transformations as trans
from pmda.hbond_analysis import HydrogenBondAnalysis as HBA


def main():
    # load Universe
    # dt in ps
    u = mda.Universe(
        "../input_data/interface.psf", "../input_data/trajectory.xyz", dt=0.025
    )
    dim = [16.869, 16.869, 41.478, 90, 90, 120]
    transform = trans.boxdimensions.set_dimensions(dim)
    u.trajectory.add_transformations(transform)
    u.transfer_to_memory()

    hbonds = HBA(
        universe=u,
        donors_sel=None,
        hydrogens_sel="name H",
        acceptors_sel="name O",
        d_a_cutoff=3.0,
        d_h_a_angle_cutoff=150,
        update_selections=False,
    )
    hbonds.run()
    return hbonds


if __name__ == "__main__":
    start = time.process_time()
    hbonds = main()
    print(hbonds.hbonds)
    fmt = "\nWork Completed! Used Time: {:.3f} seconds"
    print(fmt.format(time.process_time() - start))
