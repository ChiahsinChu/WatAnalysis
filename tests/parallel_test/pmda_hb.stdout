[[0.00000000e+00 2.22000000e+02 2.24000000e+02 6.24000000e+02
  2.88494294e+00 1.75897184e+02]
 [0.00000000e+00 2.67000000e+02 2.69000000e+02 3.72000000e+02
  2.85001150e+00 1.75725563e+02]
 [0.00000000e+00 2.73000000e+02 2.74000000e+02 2.88000000e+02
  2.80944372e+00 1.68008902e+02]
 ...
 [9.99900000e+03 3.60000000e+02 3.62000000e+02 5.10000000e+02
  2.76265996e+00 1.68040258e+02]
 [9.99900000e+03 4.14000000e+02 4.15000000e+02 5.40000000e+02
  2.81967404e+00 1.59376530e+02]
 [9.99900000e+03 4.29000000e+02 4.30000000e+02 6.57000000e+02
  2.64638307e+00 1.65959970e+02]]

Work Completed! Used Time: 73.769 seconds
